src/deployment/2_creategenesis.js

```diff
@@ -123,7 +123,7 @@ async function main() {
      * deploy proxy
      * Do not initialize directlythe proxy since we want to deploy the same code on L2 and this will alter the bytecode deployed of the proxy
      */
-    const transparentProxyFactory = await ethers.getContractFactory('TransparentUpgradeableProxy', deployer);
+    const transparentProxyFactory = await ethers.getContractFactory('@openzeppelin/contracts/proxy/transparent/TransparentUpgradeableProxy.sol:TransparentUpgradeableProxy', deployer);
```

Comment:
One of the side effects of having two compiled contracts: (one from foundry and one from hardhat) is that we need to specifiy it more in detail